core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -221,19 +224,18 @@ protected Attributes getNameResolverParams() {
 
   private static class AuthorityOverridingTransportFactory implements ClientTransportFactory {
     final ClientTransportFactory factory;
-    @Nullable final String authorityOverride;
+    final String authorityOverride;
 
     AuthorityOverridingTransportFactory(
-        ClientTransportFactory factory, @Nullable String authorityOverride) {
+        ClientTransportFactory factory, String authorityOverride) {
       this.factory = factory;
-      this.authorityOverride = authorityOverride;
+      this.authorityOverride = Preconditions.checkNotNull(authorityOverride);
```

Comment:
This needs to have a string as the second arg.  Sometimes we don't know what version of the code is running, so it can be hard to tell what variable was null.

Optional:  just use the static import of checkNotNull, which shortens the line back down.